lib/srv/alpnproxy/conn_upgrade.go

```diff
@@ -75,6 +82,51 @@ func IsALPNConnUpgradeRequired(addr string, insecure bool) bool {
 	return result
```

Comment:
Is it possible for negotiated protocol to come back empty without getting "tls: no application protocol" error?

Reply:
yes, the result is checking for empty
```
result := testConn.ConnectionState().NegotiatedProtocol == ""
```